mint-client/src/clients/user.rs

```diff
@@ -420,3 +452,190 @@ impl From<LnClientError> for ClientError {
         ClientError::LnClientError(e)
     }
 }
+
+// -> clientd
+pub struct WrappedInvoice(pub lightning_invoice::Invoice);
+
+struct InvoiceVisitor {}
+struct PegOutReqVisitor {}
+
+impl<'de> serde::de::Visitor<'de> for InvoiceVisitor {
+    type Value = WrappedInvoice;
+
+    fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+        formatter.write_str("Could not deserialize Invoice")
+    }
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        match v.parse::<lightning_invoice::Invoice>() {
+            Ok(bolt11) => Ok(WrappedInvoice(bolt11)),
+            Err(e) => Err(serde::de::Error::custom(e.to_string())),
+        }
+    }
+}
+
+impl<'de> serde::de::Visitor<'de> for PegOutReqVisitor {
```

Comment:
Doesn't need a manual serde impl but can be derived. The amount probably gave you an error, but [that can be solved like this](https://github.com/fedimint/minimint/blob/e05cecd61aac76eed1190785546aae40f60c297a/modules/minimint-wallet/src/lib.rs#L97).

Reply:
Perfect thanks